client/components/jetpack/card/jetpack-free-card-button/index.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import React, { FC } from 'react';
+import { useSelector } from 'react-redux';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { Button } from '@automattic/components';
+
+/**
+ * Internal dependencies
+ */
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import useTrackCallback from 'calypso/lib/jetpack/use-track-callback';
+import { addQueryArgs } from 'calypso/lib/route';
+import getJetpackWpAdminUrl from 'calypso/state/selectors/get-jetpack-wp-admin-url';
+import { JPC_PATH_BASE } from 'calypso/jetpack-connect/constants';
+
+/**
+ * Type dependencies
+ */
+import type { QueryArgs } from 'calypso/my-sites/plans-v2/types';
+
+interface JetpackFreeCardButton {
+	className?: string;
+	label?: TranslateResult;
+	primary?: boolean;
+	siteId: number | null;
+	urlQueryArgs: QueryArgs;
+}
+
+const JetpackFreeCardButton: FC< JetpackFreeCardButton > = ( {
+	className,
+	label,
+	primary = false,
+	siteId,
+	urlQueryArgs,
+} ) => {
+	const translate = useTranslate();
+	const wpAdminUrl = useSelector( getJetpackWpAdminUrl );
+	const onClickTrack = useTrackCallback( undefined, 'calypso_product_jpfree_click', {
+		site_id: siteId || undefined,
+	} );
+
+	// Jetpack Connect flow uses `url` instead of `site` for a site URL
+	const { site: url, ...restQueryArgs } = urlQueryArgs;
+	const startHref = isJetpackCloud()
+		? addQueryArgs( { url, ...restQueryArgs }, `https://wordpress.com${ JPC_PATH_BASE }` )
```

Comment:
For testing purposes, I recommend you to replace `https://wordpress.com` with `http://calypso.localhost:3000`. This will make the Jetpack Free button redirect you to your local Calypso Blue instead of to production.

Reply:
Is there a way to get this from config? Will this work in any Calypso environment?

Reply:
I would say that right now there is no way to get it from config, but I think you're right if you suspect that this is not the best way of doing it. Since this was a mere refactor of the `JetpackFreeCard` button and that the PR is about the issue with the product selection, I'm going to keep it the way it is for now and propose a better approach in a follow-up PR.